psl-evaluation/src/test/java/org/linqs/psl/utils/evaluation/statistics/DiscretePredictionComparatorTest.java

```diff
@@ -144,4 +144,44 @@ public void testAccuracy() {
 				assertEquals(26.0/30.0, acc, 1e-10);
 		}		
 	}
+
+	@Test
+	public void testPrecisionNegativeClass() {
+		for (double thresh = 0.1; thresh <= 1.0; thresh += 0.1) {
+			comparator.setThreshold(thresh);
+			DiscretePredictionStatistics comparison = comparator.compare(predicate, 6*5);
+			double prec = comparison.getPrecision(DiscretePredictionStatistics.BinaryClass.NEGATIVE);
+			if (thresh <= 0.8)
+				assertEquals(24.0/25.0, prec, 1e-10);
+			else
+				assertEquals(26.0/30.0, prec, 1e-10);
+		}
+	}
+	
+	@Test
+	public void testRecallNegativeClass() {
+		for (double thresh = 0.1; thresh <= 1.0; thresh += 0.1) {
+			comparator.setThreshold(thresh);
+			DiscretePredictionStatistics comparison = comparator.compare(predicate, 6*5);
```

Comment:
Where did the magic numbers "6 * 5" come from? I know the other tests used it as well, but you should probably change them all so that it is clear where they come from.